src/github_domain.py

```diff
@@ -20,41 +20,46 @@
 
 from typing import Any, Dict, List, Optional, Type
 
-DEFAULT_TIMESTAMP: datetime.datetime = (
-    datetime.datetime(datetime.MINYEAR, 1, 1, tzinfo=datetime.timezone.utc))
-
 
 class Assignee:
     """A class representing an assignee of a pull request."""
 
     def __init__(
         self,
         username: str,
-        timestamp: datetime.datetime = DEFAULT_TIMESTAMP
+        assigned_on_timestamp: Optional[datetime.datetime] = None
     ) -> None:
         """
         Args:
             username: str. GitHub username of the assignee.
-            timestamp: datetime. The time when an event happened.
+            assigned_on_timestamp: datetime|None. The time when the reviewer was
+                assigned, else None.
```

Comment:
...assigned, or None if [XYZ].

(What is XYZ? The reviewer was never assigned, or isn't assigned currently, or something else? Also it's not clear to me how this could possibly be None since then the assignee isn't an assignee. This needs to be explained more clearly, or the optional case should be removed.)

Reply:
Currently, the `Assignee` class is taking one required argument which is the `username`, and using `None` as the default value for `assigned_on_timestamp`.  The purpose of using `None` is to initialize it, nothing else. First, we are creating an `Assignee` instance in `from_github_response` and then using `update_assignee_timestamp` -> `__process_activity` -> `set_assigned_on_timestamp` to set the timestamp value. If there is an assignee only then an `Assignee` instance will be created with that username. What do you think about keeping `None` to initialize the argument with an explanation? I think making `assigned_on_timestamp` a default argument can be tricky because we need to make separate API calls to fetch specific PR timelines. FYI, Previously we were using `DEFAULT_TIMESTAMP: datetime.datetime = (
    datetime.datetime(datetime.MINYEAR, 1, 1, tzinfo=datetime.timezone.utc))` to initialize. I am ok with any of them but `None` is making more sense to me.

Reply:
It would be better if we initialized it on creation. Can we get all the necessary info and only then create the Assignee object?

Reply:
Yeah, I think we can do that by modifying the functions in `github_services`. Signing off for the night, will try to do it tomorrow morning.

Reply:
I have updated and removed a few functions and methods and now we are creating the Assignee object at once.